### PR TITLE
[sw] Clean up most uses of uart_send_* into LOG_*

### PR DESCRIPTION
**Pull request by mcy**
Some uses remain which don't quite cleanly fit into the purpose that the LOG macros serve, mostly in the PASS/FAIL domain.

### CODE REVIEW

File: sw/device/lib/handler.c [hunk 1]
```diff
@@ -13,31 +14,16 @@
  */
 static uint32_t get_mtval(void) {
   uint32_t mtval;
-  asm volatile("csrr %0, mtval" : "=r"(mtval) : :);
+  asm volatile("csrr %0, mtval" : "=r"(mtval));
```
Comment (gkelly) [inline sw/device/lib/handler.c]: This should be a separate commit.

File: sw/device/lib/handler.c [hunk 2]
```diff
@@ -13,31 +14,16 @@
...
-  };
-}
-
 // Below functions are default weak exception handlers meant to be overriden
 __attribute__((weak)) void handler_exception(void) {
   uint32_t mcause;
   exc_id_t exc_cause;
 
-  asm volatile("csrr %0 , mcause" : "=r"(mcause) : :);
+  asm volatile("csrr %0 , mcause" : "=r"(mcause));
```
Comment (gkelly) [inline sw/device/lib/handler.c]: Same commit as the other asm change.  Also insert space between `"=r"` and `(mcause)`.

File: sw/device/lib/handler.c (hunk 1 above)
Comment (mcy) [reply sw/device/lib/handler.c]: Ok, I'll do those in a followup PR at some point.

**Comment by sriyerg** [comment]
Looks like the private CI failed due to exceeding 15 minute run time limit. I have seen this happen on rare occasions. I have re-triggered the check now. 

Typically the DV regressions complete within anywhere between 7 to 12 minutes. Due to heavy randomization on several fronts, it is hard to contain the sim time within s deterministic limit. I will update the limit to 20 minutes to be on the safe side.

File: sw/device/examples/hello_usbdev/hello_usbdev.c [hunk 3]
```diff
@@ -82,6 +83,8 @@ static dif_gpio_t gpio;
 
 int main(int argc, char **argv) {
   uart_init(kUartBaudrate);
+  base_set_stdout(uart_stdout);
```
Comment (sriyerg) [inline sw/device/examples/hello_usbdev/hello_usbdev.c]: Can this be placed inside `base_log_internal_core()` instead?

File: sw/device/tests/flash_ctrl/flash_test.c [hunk 4]
```diff
@@ -3,23 +3,24 @@
...
-      uart_send_str("Assertion failed at line 0x"); \
-      uart_send_uint(__LINE__, 32);                 \
-      uart_send_str("\r\nFAIL!\r\n");               \
-      abort();                                      \
-    }                                               \
+#define CHECK(cond)                   \
+  do {                                \
+    if (!(cond)) {                    \
+      LOG_ERROR("Assertion failed!"); \
+      uart_send_str("FAIL!\r\n");     \
```
Comment (sriyerg) [inline sw/device/tests/flash_ctrl/flash_test.c]: LOG_INFO?

File: sw/device/examples/hello_usbdev/hello_usbdev.c (hunk 3 above)
Comment (mcy) [reply sw/device/examples/hello_usbdev/hello_usbdev.c]: No. The logging functions are output agnostic, so it's the binary's responsibility to set up a stdout. By default, logging does nothing, after all.

**Comment by gkelly** [comment]
Hey @mcy, this looks pretty useful. It's also got all the approvals needed by the looks of it. Can you implement Sri's suggestion and rebase?

**Comment by mcy** [comment]
This PR is basically irrelevant at this point.
